### PR TITLE
[3.9] Namespace filesystem

### PR DESCRIPTION
**Pull request by fastslack**
Filesystem namespace

### Summary of Changes

Moved all libraries names to use namespace

### Testing Instructions

Use as usual, all should work fine.

### Expected result

No errors

### CODE REVIEW

**Comment by laoneo** [comment]
Thank you very much. The wrappers can't be deleted on Joomla 3.9, after this one gets merged, we can then delete them in the 4.0 branch as we did with the other wrappers. I guess for the rest we can slowly migrate the code then to the framework filesystem package internally.

File: libraries/src/Filesystem/Patcher.php [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+defined('JPATH_PLATFORM') or die;
 
 /**
  * A Unified Diff Format Patcher class
  *
  * @link   http://sourceforge.net/projects/phppatcher/ This has been derived from the PhpPatcher version 0.1.1 written by Giuseppe Mazzotta
  * @since  12.1
  */
-class JFilesystemPatcher
+class Patcher
```
Comment (C-Lodder) [inline libraries/src/Filesystem/Patcher.php:22]: Should be `FilesystemPatcher`?

Comment (fastslack) [reply libraries/src/Filesystem/Patcher.php:22]: Hmm i dont know, i think the correct way to call this is ```use Joomla\CMS\Filesystem\Patcher;``` and not ```use Joomla\CMS\Filesystem\FilesystemPatcher;```

Comment (C-Lodder) [reply libraries/src/Filesystem/Patcher.php:22]: I dont mean the `use` statement. I meant the class name. You've used `class Patcher` ans it should be `class FileSystemPatcher`

Comment (mbabker) [reply libraries/src/Filesystem/Patcher.php:22]: No, it's fine as is

**Comment by fastslack** [comment]
Closing this, i ll open a new one soon with wrappers

**Comment by laoneo** [comment]
Or just add the files again and commit them to the branch then you should be good to go, no need to close it.

**Comment by fastslack** [comment]
But if i add the files i ll lose all git logs, also i need to change others files that point to wrappers. Dont worry i can do it fast using the patches

**Comment by laoneo** [comment]
Not sure if your really loose the logs. Which files have been using the wrappers. Probably you can leave them as we are going to remove the wrappers in 4 anyway.

**Comment by fastslack** [comment]
Its done here: https://github.com/joomla/joomla-cms/pull/18584

**Pull request closed without merging**